themes/socialbase/templates/profile/profile--compact.html.twig

```diff
@@ -20,5 +20,7 @@
  */
 #}
 <div class="avatar">
-  {{ content }}
+  <a href="{{ profile_stream_url }}" title="{{ profile_name|render }}">
```

Comment:
This now breaks the widget settings "link to content" for site builders (perhaps some site had it disabled). Is it not possible to add the "#title" of the URL in a preprocess hook if the URL is set?

Reply:
There was code in .theme file, where all settings "link to content" was overrid like  link. Also this method used in some another place for profile template. If for some site need disable it - need only orevvide template and remove link.

Reply:
I had hoped the following snippet would work:

```php
    // Only change the URL if it's already linked to content. This ensures that
    // the widget setting "Link to content" remains functional.
    if (!empty($build['field_profile_image'][0]['#url'])) {
      $uid = $entity->getOwnerId();
      $name = $entity->getOwner()->getDisplayName();

      // Get the url from the user profile.
      $url = Url::fromRoute('entity.user.canonical', [
        'user' => $uid,
      ], [
        'attributes' => [
          'title' => $name,
        ]
      ]);

      // Add it to the build.
      $build['field_profile_image'][0]['#url'] = $url;
    }
  }
```

It would be a more elegant solution, however it runs into https://www.drupal.org/project/drupal/issues/2829990

Reply:
This doens't work so easy. If you set url for image - then for image drupal use "image-formatter.html.twig" template and there set link with image.

Reply:
We could change the image formatter template to copy the alt/title from the image onto the URL?

Reply:
Yes, we can set alt from image like title attribute for link, but for now "Alternative text" aren't mandatory for profile image and can be not so clear as user name. If this is okay - I can change this like you say.

Reply:
I suppose for now we can just leave it as you made it. Maybe once the Drupal Core issue is fixed we can come back to a different solution

Reply:
Please see comment here: https://www.drupal.org/project/social/issues/2938820#comment-12523445
If you have agreed to leave the current commit as is (with the anchor and its title attribute) then can I suggest we go further and add the alt tag this way too?
I think the alt tag is more important than the title (especially since this is an accessibility-focused PR)

eg:   

social/themes/socialbase/templates/profile/profile--compact.html.twig
<code>
<a href="{{ profile_stream_url }}">
    <img src="{{ profile_image_url }}" width="44" height="44"
         class="img-responsive" alt="{{ profile_name }}{{ ' profile image'|t
    }}" title="{{ profile_name }}">
  </a>
</code>

/social/modules/social_features/social_profile/social_profile.module
<code>
  // Add the profile image.
  if (!empty($profile->field_profile_image->entity)) {
    $variables['profile_image_url'] = ImageStyle::load('social_medium')
      ->buildUrl($profile->field_profile_image->entity->getFileUri());
  }
</code>

Can you see issues with this approach?

Reply:
I don't like having to set the image dimensions

Reply:
@left23 For now image has alternative text as well, only need set this for user, because we doesn't have alternative text for images as mandatory and this is empty. So we will have the both title for link and alternative text for image.

Reply:
Ok, thanks for explaining further. 
It seems like a blank alt field is almost as ineffective as no alt tag. Same result.
How do you propose we set this for the user?

Reply:
@left23 I think we need set alternative text for all image as mandatory for Open Social. But maybe for this need separate task.